src/libostree/ostree-sysroot-deploy.c

```diff
@@ -1792,14 +1792,78 @@ bootloader_entry_filename (OstreeSysroot *sysroot, guint n_deployments,
   if (use_old_naming)
     {
       const char *stateroot = ostree_deployment_get_osname (deployment);
-      return g_strdup_printf ("ostree-%d-%s.conf", index, stateroot);
+      return g_strdup_printf ("ostree-%d-%s", index, stateroot);
     }
   else
     {
-      return g_strdup_printf ("ostree-%d.conf", index);
+      return g_strdup_printf ("ostree-%d", index);
     }
 }
 
+#ifdef ENABLE_BOOT_COUNT
+
+#define BOOT_COUNT_MAX_RETRIES 3
+
+static gint
+bootloader_get_max_boot_tries (OstreeSysroot *self, OstreeDeployment *deployment,
+                               GCancellable *cancellable, GError **error)
+{
+  g_autofree char *tries_file_path = ostree_deployment_get_boot_tries_relpath (deployment);
```

Comment:
Hmmm...one super messy thing here is that at this point we may not have actually written out the deployment's `/etc`. We really want a version of this that's in `/usr`.

AFAICS, the uapi standard doesn't specify `/etc/kernel/tries` - that's an invention of `kernel-install`.

To combine with "make this configurable via sysroot" we could have e.g.:

```
[sysroot]
boot_counting=3
```

But OTOH I've been trying to get away from the sysroot configuration for other reasons...so we may need to invent `/usr/lib/ostree/sysroot.conf` or so as a pair with `/usr/lib/ostree/prepare-root.conf`?

Reply:
added support for `boot_counting` property in repo config